framework/web/CController.php

```diff
@@ -390,9 +390,10 @@ public function processDynamicOutput($output)
 	protected function replaceDynamicOutput($matches)
 	{
 		$content=$matches[0];
-		if(isset($this->_dynamicOutput[$matches[1]]))
+		if(isset($this->_dynamicOutput[$matches[1]]) && is_array($this->_dynamicOutput[$matches[1]]))
```

Comment:
> How does it related to #3846?

As I've mentioned before, that is from another pending pull-request #3788. It is not related to this issue, it is here just because I did not find another way to contribute having pending pull-request at `master` branch. I guess the pull-request should be re-rolled - but at the time I don't see a chance. Please re-roll yourself as my `master` branch already has that commit and I can't create a fork from scratch.